### PR TITLE
Add framework weak linking setting

### PR DESCRIPTION
**Pull request by alvarhansen**
As far as I can find, there are 2 possible values for this: `Required` and `Weak`. Specifying required is optional as it is the default value.

In Xcode this setting is available in multiple places. For example in Project general settings:

Apple documentation on weak linking: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/WeakLinking.html

### CODE REVIEW

File: Sources/XcodeGenKit/PBXProjGenerator.swift [hunk 1]
```diff
@@ -459,6 +459,14 @@ public class PBXProjGenerator {
             return ["ATTRIBUTES": embedAttributes]
         }
 
+        func getLinkingSettings(dependency: Dependency) -> [String: Any] {
```
[the remark this answers is not in the document]
Comment (alvarhansen) [reply Sources/XcodeGenKit/PBXProjGenerator.swift]: Renamed to `getDependencyFrameworkSettings`

File: Sources/XcodeGenKit/PBXProjGenerator.swift [hunk 2]
```diff
@@ -459,6 +459,14 @@ public class PBXProjGenerator {
...
+            var linkingAttributes: [String] = []
+            if dependency.weakLink {
+                linkingAttributes.append("Weak")
+            }
+            return ["ATTRIBUTES": linkingAttributes]
```
[the remark this answers is not in the document]
Comment (alvarhansen) [reply Sources/XcodeGenKit/PBXProjGenerator.swift]: Changed function return type to optional and return nil when `linkingAttributes` is empty.

**Comment by alvarhansen** [comment]
@yonaskolb I added few tests and created new fixture.

I tried out the `-weak_framework` linker flag, and it seems it does the same thing.
https://github.com/alvarhansen/FrameworkLinkingTest/commits/master
I tested it by running `nm -mg FrameworkLinkingTest.app/FrameworkLinkingTest | grep MyFramework`
For non weak linking the result is:
```
(undefined) external _$S11MyFramework3FooCACycfC (from MyFramework)
(undefined) external _$S11MyFramework3FooCMa (from MyFramework)
```

For both weak linking approaches the result is:
```
(undefined) weak external _$S11MyFramework3FooCACycfC (from MyFramework)
(undefined) weak external _$S11MyFramework3FooCMa (from MyFramework)
```

**Comment by yonaskolb** [comment]
Awesome, great work @alvarhansen! 👏

**Pull request merged**